src/codegen/doc/format.rs

```diff
@@ -77,32 +77,51 @@ fn format(mut input: &str, symbols: &symbols::Info) -> String {
 }
 
 static SYMBOL: Lazy<Regex> =
-    Lazy::new(|| Regex::new(r"(^|[^\\])[@#%]([\w]+\b)([:.]+[\w_-]+\b)?").unwrap());
+    Lazy::new(|| Regex::new(r"(^|[^\\])([@#%])(\w+\b)([:.]+[\w-]+\b)?").unwrap());
 static FUNCTION: Lazy<Regex> = Lazy::new(|| Regex::new(r"(\b[a-z0-9_]+)\(\)").unwrap());
-static GDK_GTK: Lazy<Regex> = Lazy::new(|| Regex::new(r"G[dt]k[A-Z][\w]+\b").unwrap());
+static GDK_GTK: Lazy<Regex> = Lazy::new(|| Regex::new(r"G[dt]k[A-Z]\w+\b").unwrap());
 static TAGS: Lazy<Regex> = Lazy::new(|| Regex::new(r"<[\w/-]+>").unwrap());
-static SPACES: Lazy<Regex> = Lazy::new(|| Regex::new(r"[ ][ ]+").unwrap());
+static SPACES: Lazy<Regex> = Lazy::new(|| Regex::new(r"[ ]{2,}").unwrap());
 
 fn replace_c_types(entry: &str, symbols: &symbols::Info) -> String {
-    let lookup = |s: &str| -> String {
-        symbols
-            .by_c_name(s)
-            .map(symbols::Symbol::full_rust_name)
-            .unwrap_or_else(|| s.into())
-    };
+    let lookup_opt =
+        |s: &str| -> Option<String> { symbols.by_c_name(s).map(symbols::Symbol::full_rust_name) };
+    let lookup = |s: &str| -> String { lookup_opt(s).unwrap_or_else(|| s.into()) };
+
     let out = SYMBOL.replace_all(entry, |caps: &Captures<'_>| {
-        format!(
-            "{}`{}{}`",
-            &caps[1],
-            lookup(&caps[2]),
-            caps.get(3).map(|m| m.as_str()).unwrap_or("")
-        )
+        let trail = caps.get(4).map(|m| m.as_str()).unwrap_or("");
+        match &caps[2] {
+            // Parameter (and member?) names
```

Comment:
I don't really know what `@/#/%` mean _exactly_ just yet. Is it a doxygen thing?

Reply:
`@` is struct member/parameter/enum value in struct/function/enum documentation. `#` is for referencing types, `%` for referencing constants and enum values. It's a gtk-doc thing.

Reply:
Right. `@` is a tricky one because we _can_ reference enum variants, but not parameters nor struct members. Perhaps the same `if let Some(name) = lookup_opt()` can be used here to tell whether it is referencing a valid C symbol or not.

Reply:
By the way it appears C code mixes and matches `@#%` for "global names" like (enum) constants, functions, and types. I guess `@` simply means "associated to" because it is also used to reference functions on the same type?

If we have a valid C symbol for it, perhaps this should emit Self::<member>` instead of `crate::<type>::<member>`.

Reply:
> but not parameters nor struct members.

Not reference, but usually these are used to document those directly and then you would put the doc comment directly on that parameter or struct member or not?

Reply:
And for mixed usage in C code, gtk-doc is very forgiving with making mistakes and I think nobody (including me) knows how these things would be done correctly :)

Reply:
> Not reference, but usually these are used to document those directly and then you would put the doc comment directly on that parameter or struct member or not?

I've previously seen a list of ` * @GST_ENUM_VARIANT: the docs` as a list on top of the type, rather than on the variant directly. Not sure how that works for struct members. Anyway, this issue is about _linking to_ members and fields from the documentation itself, say if you have a large guiding paragraph on some type explaining how to use and what members to look at. You don't want sentences cut out and moved to the member directly :wink: 

Besides, structs are less relevant for GStreamer-rs since FFI fields are generally hidden from the user.

> And for mixed usage in C code, gtk-doc is very forgiving with making mistakes and I think nobody (including me) knows how these things would be done correctly :)

:/ that makes it all the tougher to do these links correctly, and leads us back to the question in the last paragraph of the PR description: how would we land this? Accept that there are many broken links? Try to filter them out or tell rustdoc to omit `[]` - if at all possible?

Reply:
Okay, gave `Self::<memb>` a try since we have neat access to 

- `@GstGLContext`/`@GstRTSPSessionMedia`: We have references directly to `Self`, is that correct or should these use `#`? There are not too many...;
- `@gst_video_decoder_set_output_state()` on `VideoCodecState` (needs upstream fixing, `@` should be dropped);

Other than that I might almost submit a minor PR that comes before this taking care of just `Self` links, way less issues :tada: